wp-modules/editor/model.php

```diff
@@ -116,16 +120,17 @@ function save_metadata_to_pattern_file( $override, $post_id, $meta_key, $meta_va
 				'post_name' => $meta_value,
 			]
 		);
+	}
 
-		if ( $pattern_name !== $meta_value ) {
-			delete_pattern( $pattern_name );
-		}
+	if ( $name_changed ) {
+		delete_pattern( $pattern_name );
 	}
 
 	return update_pattern(
 		array_merge(
 			get_pattern_defaults(),
 			$pattern ? $pattern : [ 'title' => $post->post_title ],
+			$name_changed ? [ 'slug' => $name ] : [],
```

Comment:
If they edit the title, `$name_changed` will be `true`.

And this will set the slug to the name.